src/vm/jitinterface.cpp

```diff
@@ -12570,8 +12572,16 @@ CorJitResult CallCompileMethodWithSEHWrapper(EEJitManager *jitMgr,
         flags.Set(CORJIT_FLAGS::CORJIT_FLAG_FRAMED);
     if (g_pConfig->JitAlignLoops())
         flags.Set(CORJIT_FLAGS::CORJIT_FLAG_ALIGN_LOOPS);
-    if (ftn->IsVersionableWithJumpStamp() || g_pConfig->AddRejitNops())
+
+    // Only set CORJIT_FLAG_PROF_REJIT_NOPS when we aren't versionable using 
+    // the Tiering mechanism and we are versionable using a jump stamp.
+    //
+    if ((!ftn->IsVersionableWithoutJumpStamp() && ftn->IsVersionableWithJumpStamp())
+        || g_pConfig->AddRejitNops())
+    {
         flags.Set(CORJIT_FLAGS::CORJIT_FLAG_PROF_REJIT_NOPS);
```

Comment:
This flag is always set during crossgen in `ZapInfo::ComputeJitFlags`, Do you need to do something about that too?

Reply:
Yes, This version is not working correctly and is always producing nop pads.